nautobot/core/api/schema.py

```diff
@@ -107,25 +174,47 @@ def get_writable_class(self, serializer):
             elif isinstance(child, ManyRelatedField) and isinstance(child.child_relation, SerializedPKRelatedField):
                 properties[child_name] = None
 
+        if bulk:
+            # The "id" field is always different in bulk serializers
+            properties["id"] = None
+
         if not properties:
             # There's nothing about this serializer that requires a special WritableSerializer class to be defined.
             return None
 
-        # Have we already created a WritableFooSerializer class or do we need to do so now?
-        if not isinstance(serializer, tuple(self.writable_serializers)):
+        # Have we already created a [Bulk]WritableFooSerializer class or do we need to do so now?
+        writable_name = "Writable" + type(serializer).__name__
+        if bulk:
+            writable_name = f"Bulk{writable_name}"
+        if writable_name not in self.writable_serializers:
             # We need to create a new class to use
-            writable_name = "Writable" + type(serializer).__name__
             # If the original serializer class has a Meta, make sure we set Meta.ref_name appropriately
             meta_class = getattr(type(serializer), "Meta", None)
             if meta_class:
                 ref_name = "Writable" + self.get_serializer_ref_name(serializer)
+                if bulk:
+                    ref_name = f"Bulk{ref_name}"
                 writable_meta = type("Meta", (meta_class,), {"ref_name": ref_name})
                 properties["Meta"] = writable_meta
 
-            # Define and cache a new WritableFooSerializer class
-            self.writable_serializers[type(serializer)] = type(writable_name, (type(serializer),), properties)
+            # Define and cache a new [Bulk]WritableFooSerializer class
+            if bulk:
+
+                def get_fields(self):
+                    """For Nautobot's bulk_update/partial_update/delete APIs, the `id` field is mandatory."""
+                    new_fields = {}
+                    for name, field in type(serializer)().get_fields().items():
+                        if name == "id":
+                            field.read_only = False
+                            field.required = True
+                        new_fields[name] = field
+                    return new_fields
 
-        writable_class = self.writable_serializers[type(serializer)]
+                properties["get_fields"] = get_fields
+
+            self.writable_serializers[writable_name] = type(writable_name, (type(serializer),), properties)
+
+        writable_class = self.writable_serializers[writable_name]
```

Comment:
I hate that we have to do all of this and if there's any reasonable expectation that we can undo this in v2, can we please add "v2 TODO" for these?

Reply:
I'm definitely open to ideas on how we could make the bulk and non-bulk serializers more similar. I don't see an obvious path to doing so though.